### PR TITLE
feat(koji): add koji

### PR DESCRIPTION
**Pull request by its-danny**
Not a clue what the requirements are for getting a package added, but would be nice to get my little conventional commit util in here if possible. No windows support atm, so no installer for it.

The project for reference: https://github.com/its-danny/koji

### CODE REVIEW

**Comment by coolaj86** [comment]
Hey, just wanted to let you know that I appreciate this and I'll be taking a look at it within the next week or so.

I just spent a bunch of time on Webi PRs and updates a few weeks ago and I've needed to focus on some other projects for a bit to get caught up on them.

:)

File: koji/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+homepage: https://github.com/its-danny/koji
+tagline: |
+  🦊 An interactive CLI for creating conventional commits. 
+---
+
+To update or switch versions, run `webi koji@stable` (or `@v2`, `@beta`, etc).
+
+## How to run `koji`
+
+```bash
```
Comment (coolaj86) [inline koji/README.md:12]: Each command should be in its own code block and have a short description.

File: koji/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+koji
+
+# with emoji
+koji --emoji
```
Comment (coolaj86) [inline koji/README.md:16]: I want a little more detail. I can't tell from this alone how to use this command. Is it a substitute for `git commit`?

Or do I run `git commit -m "do some stuff whatever"` and then this adds the prefix?

File: koji/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+# as a git hook (prepare-commit-msg)
+koji --hook
+```
```
Comment (coolaj86) [inline koji/README.md:20]: Again, I just need a little more info. Most people who come to webi are very much _NOT_ expert developers and probably wouldn't understand.

I'm guessing this runs whenever you run `git commit` with any options and rejects messages that don't include commits? Or asks you to add a prefix?

Just needs more detail. :)

File: koji/install.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    chmod a+x "$pkg_src_cmd"
+}
+
+function pkg_link() {
+    # rm -f "$HOME/.local/bin/koji"
+    rm -f "$pkg_dst_cmd"
+
+    # ln -s "$HOME/.local/opt/koji-1.3.4/bin/koji" "$HOME/.local/bin/koji"
+    ln -s "$pkg_src_cmd" "$pkg_dst_cmd"
+}
```
Comment (coolaj86) [inline koji/install.sh:36]: Please run the code formatting tools before committing:

```bash
webi prettier
webi shfmt

# npx prettier@2 -w '**/*.{js,md,html}'
npm run prettier 

# shfmt -w -i 4 -sr -s ./
npm run shfmt
```

**Review by coolaj86 — CHANGES_REQUESTED** [review]
This looks really awesome! It's up on beta at https://beta.webinstall.dev/koji

I bet @wmerfalen will love it (and we'll probably start using it at work).

That said, I have a few requests before merging this in. Mostly because webinstall.dev is meant for idiots like me and the docs aren't quite complete enough for me to know how I'm supposed to use this.

I'd love to get this into prod in the next day or two if you'd just make the couple of requested updates.

:)

File: koji/README.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+---
+title: koji
+homepage: https://github.com/its-danny/koji
+tagline: |
+  🦊 An interactive CLI for creating conventional commits. 
+---
+
+To update or switch versions, run `webi koji@stable` (or `@v2`, `@beta`, etc).
+
```
Comment (coolaj86) [inline koji/README.md:9]: Please add a "Files" section as the first subsection of the Cheat Sheet, as seen here:
- https://webinstall.dev/gpg-pubkey/
- https://webinstall.dev/vim-shfmt/
- https://webinstall.dev/delta/

**Comment by coolaj86** [comment]
P.S. Would you like to join us in remote for a lightning talk in February at [Utah Rust](https://www.meetup.com/utah-rust/events/)? Someone there may be able to help with Windows support.

**Comment by its-danny** [comment]
@coolaj86 heyo, will get to this tonight! i plan on adding windows support at some point, mostly comes down to not knowing how to have github actions build for windows yet but its on my todo list.

**Comment by its-danny** [comment]
@coolaj86 I think all is well, fleshed out README and ran prettier & shfmt.

File: koji/README.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+[conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), ran in
+place of `git commit`.
+
+### Using koji
+
+```bash
+cd dev/work-stuff
+# Do some work
+cd dev/work-stuff
+git add .env.production
```
Comment (coolaj86) [inline koji/README.md:34]: umm... no... DO NOT commit private keys for prod!!!

**Comment by coolaj86** [comment]
Now that I understand this better and I've had a chance to try it out, I've updated the README in a way that I feel is more targeted towards our audience.

**Review by coolaj86 — APPROVED** [review]

**Comment by coolaj86** [comment]
Live at https://webinstall.dev/koji/

I don't have it on the homepage yet, but I think it needs a little work before it's ready for prime time. :)

File: koji/README.md (hunk 6 above)
Comment (its-danny) [reply koji/README.md]: joke from repo README lol, probably shoulda changed it

**Pull request merged**